subworkflows/nf-core/dia_proteomics_analysis/main.nf

```diff
@@ -0,0 +1,426 @@
+        }                                                                      // [meta_exp_searchdb, logSettings, empirical_library]
+
+    //
+    // Generate mass accuracy settings combining the settings from the log and the settings from the input
+    //
+
+    ch_individual_analysis_input = input.mass_tolerance_settings // [meta_input_fasta, precursor_tolerance, fragment_tolerance, precursor_tolerance_unit, fragment_tolerance_unit, ms_file, fasta]
+        .map{ [it[0].experiment] + it } // [meta_exp_searchdb, meta_input_fasta, precursor_tolerance, fragment_tolerance, precursor_tolerance_unit, fragment_tolerance_unit, ms_file, fasta]
+        .combine(ch_empirical_output, by: 0) // [meta_exp_searchdb, meta_input_fasta, precursor_tolerance, fragment_tolerance, precursor_tolerance_unit, fragment_tolerance_unit, ms_file, fasta, logSettings, empirical_library]
+        .map { meta_exp, meta_input_fasta, precursor_tolerance, fragment_tolerance, precursor_tolerance_unit, fragment_tolerance_unit, ms_file, fasta, logSettings, empirical_library ->
+            def mass_settings = defineMassAccuracySettings(logSettings, precursor_tolerance, fragment_tolerance, precursor_tolerance_unit, fragment_tolerance_unit, wf_scan_window, wf_mass_acc_automatic, wf_scan_window_automatic, wf_pg_level)
+            def meta_with_settings = meta_input_fasta + mass_settings
+            [meta_with_settings, ms_file, fasta, empirical_library]  
+        }
+        
+    //
+    // MODULE: Individual analysis
+    //
+
+    DIANN_INDIVIDUALANALYSIS(
+        ch_individual_analysis_input.map { meta, ms_file, fasta, empirical_library ->
+            [meta, ms_file, [], fasta, empirical_library, []]  // DIANN module input: [meta, ms_files, ms_file_names, fasta, library, quant]
+        }
+    )
+    ch_versions = ch_versions.mix(DIANN_INDIVIDUALANALYSIS.out.versions)
+
+    //
+    // MODULE: Final quantification
+    //
+
+    ch_final_quantification_input = ch_individual_analysis_input // [meta_with_settings, ms_file, fasta, empirical_library]
+        .join(DIANN_INDIVIDUALANALYSIS.out.diann_quant)          // [meta_with_settings, ms_file, fasta, empirical_library, diann_quant]
+        .map { [it[0].experiment] + it.drop(1)}                  // [meta_exp_searchdb, ms_file, fasta, empirical_library, diann_quant]
+        .groupTuple()                                            // [meta_exp_searchdb, [ms_file], [fasta], [empirical_library], [diann_quant]]
+        .map{ sortListsByPathName(it, 1) }                       // [meta_exp_searchdb, [sorted_ms_file], [sorted_fasta], [sorted_empirical_library], [sorted_diann_quant]]
+        .map { meta, ms_files, fasta, empirical_library, diann_quant ->
+            [meta, [], ms_files.collect{ it.name}, fasta, empirical_library, diann_quant] // Use ms_file_names instead of ms_files for final quant
+        }
+
+    DIANN_FINALQUANTIFICATION(ch_final_quantification_input)
```

Comment:
Does this process quantify across runs?

Reply:
Yes, the same as the quantms workflow. The groupTuple in the ops above is how we merge all the inputs.